src/lib/y2partitioner/widgets/encrypt_password.rb

```diff
@@ -14,34 +14,26 @@ def initialize(controller)
         @checker = Y2Storage::EncryptPasswordChecker.new
       end
 
-      # @macro seeAbstractWidget
-      def help
-        # help text for cryptofs
-        _(
-          "<p>\n" \
-            "You will need to enter your encryption password.\n" \
-          "</p>\n" \
-          "<p>\n" \
-            "If the encrypted file system does not contain any system file and therefore is\n" \
-            "not needed for the update, you may select <b>Skip</b>. In this case, the\n" \
-            "file system is not accessed during update.\n" \
-          "</p>\n"
-        )
-      end
-
       # @macro seeAbstractWidget
       def validate
-        msg = checker.error_msg(pw1, pw2)
-        return true unless msg
+        return true if random_password?
+
+        error_msg = checker.error_msg(pw1, pw2)
+        return true unless error_msg
 
-        Yast::Report.Error(msg)
+        Yast::Report.Error(error_msg)
         Yast::UI.SetFocus(Id(:pw1))
+
         false
       end
 
       # @macro seeAbstractWidget
       def store
-        @controller.encrypt_password = pw1
+        if random_password?
+          @controller.random_password = true
+        else
+          @controller.encrypt_password = pw1
```

Comment:
do you test it if you are going back? I suspect that as random password and also encrypt_password is not reset, that it would still contain old value.

Reply:
To be honest, no, I did not test that scenario. But I just tested it now and what I see is that the described problem could not be reproduce, unless I am wrong.

 I guess that does not happen because this piece of code is located in the `#store` method, which seems to be the [second to last called in the widget's "workflow"](https://github.com/yast/yast-yast2/blob/e5231602858c6353d3802b09ff7a6b3d6e64255f/library/cwm/src/lib/cwm/abstract_widget.rb#L162). So, at the moment that user decides to go back, the `#store` method it has not been executed yet.

What do you think? Am I right?

Reply:
@dgdavid well, store is called when you leaving this dialog. If there is more actions, then when going back from next dialog it will be loaded and stored again. So I am talking about scenario where you ahve this dialog lets call it A and then another following which have back button, lets call it B. 

And not try this scenario:
A -> B (back) -> A
and in this case I believe that 
A does not have previous values ( due to hardcoded button box )
A have problem in this store when going again to B as then controller will have both random password and encrypt password.